portal/src/graphql/adminapi/UsersScreen.tsx

```diff
@@ -83,33 +167,125 @@ const UsersScreen: React.FC = function UsersScreen() {
     setOffset(offset);
   }, []);
 
-  const { loading, error, data, refetch } = useQuery<
-    UsersListQuery,
-    UsersListQueryVariables
-  >(query, {
-    variables: {
-      pageSize,
-      cursor,
-    },
-    fetchPolicy: "network-only",
-  });
+  const listQuery = useLazyQuery<UsersListQuery, UsersListQueryVariables>(
+    LIST_QUERY,
+    {
+      fetchPolicy: "network-only",
+    }
+  );
+
+  const searchQuery = useLazyQuery<UsersSearchQuery, UsersSearchQueryVariables>(
+    SEARCH_QUERY,
+    {
+      fetchPolicy: "network-only",
+    }
+  );
+
+  let execute: (options?: QueryLazyOptions<UsersSearchQueryVariables>) => void;
+  let refetch: (() => void) | undefined;
+  let loading: boolean;
+  let error: unknown;
+  let data: UsersListQuery | undefined;
+  if (searchKeyword !== "") {
+    execute = searchQuery[0];
+    data = searchQuery[1].data;
+    refetch = searchQuery[1].refetch;
+    loading = searchQuery[1].loading;
+    error = searchQuery[1].error;
+  } else {
+    execute = listQuery[0];
+    data = listQuery[1].data;
+    refetch = listQuery[1].refetch;
+    loading = listQuery[1].loading;
+    error = listQuery[1].error;
+  }
 
   const prevDataRef = useRef<UsersListQuery | undefined>();
   useEffect(() => {
     prevDataRef.current = data;
   });
   const prevData = prevDataRef.current;
 
-  const messageBar = useMemo(
-    () => error && <ShowError error={error} onRetry={refetch} />,
-    [error, refetch]
+  const search = useCallback(() => {
+    execute({
+      variables: {
+        sortBy,
+        sortDirection,
+        searchKeyword,
+        pageSize,
+        cursor,
+      },
+    });
+  }, [execute, sortBy, sortDirection, searchKeyword, cursor]);
+
+  // Initial execute
+  useEffect(() => {
+    search();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
I found that the user list page would flicker when I land on the user list page, is it the problem that you mentioned? I checked a bit and here are some of the observations. 

1. We request the graphql server twice when we land on the user page.
2. `useEffect` will be called after the first render. We have 3 `useEffect` to control calling `search`, so `search` was called 3 times when we landed on the page.
3. From point 2, graphql server should be called 3 times but not twice? I found that since the first two `search` call the timing is too close, we had called  `execute` twice but it only fire one graphql request (apollo handled it?). And the third `search` was called after 500ms, so it fire another graphql request. So the page flickers.

Maybe we should update to add some logic to avoid the second and third `useEffect` call `search` in the first render?

ref: https://reactjs.org/docs/hooks-effect.html
> Does useEffect run after every render? Yes! By default, it runs both after the first render and after every update.